config/internal/configsource/README.md

```diff
@@ -0,0 +1,149 @@
+                                                                                                              
+                                                                                                                                                                                                                            
+```
+
+The `Resolve` method proceeds in the following steps:
+
+1. Create the `configsource.ConfigSource` objects defined the `config_sources` section of the initial configuration;
+2. For each config node (key) of the initial configuration:
+    1. Skip config node if it is under the `config_sources` section; 
+    2. Parse the node value transforming any config source invocation, or environment variable, into the retrieved data;
+    3. Add the key and the value retrieved above into the resulting configuration;
+3. Return the resulting, aka effective, configuration.
+
+### Processing Config Source Invocations
+
+For each config source invocation, e.g. `${include:/cfgs/rcvrs/use.yaml}`, the code proceeds as in the following steps:
+
+1. Find the corresponding `configsource.ConfigSource` object by its name, given in the initial configuration under the `config_sources` section;
+2. Get, or create a new, `configsource.Session` from the `configsource.ConfigSource` object;
+3. Use the `configsource.Session` to retrieve the data according to the parameters given in the invocation;
+
+```terminal
++---------+         +--------------+         +---------+          +-----------+
+| Manager |         | ConfigSource |         | Session |          | Retrieved |
++---------+         +--------------+         +---------+          +-----------+
+    |                      |                      |                     |      
+    |      NewSession      |                      |                     |      
+    |--------------------->+--+                   |                     |      
+    |                      |  |                   |                     |      
+    |                      |  |------------------>+--+                  |      
+    |                      |  |                   |  |                  |      
+    |                      |  |                   |  |                  |      
+    |                      |  |<------------------+--+                  |      
+    |                      |  |                   |                     |      
+    |<---------------------+--+                   |                     |      
+    |  "Session object"    |                      |                     |      
+    |                      |                      |                     |      
+    |                      |       Retrieve       |                     |      
+    |-------------------------------------------->+--+                  |      
+    |                      |                      |  |   NewRetrieved   |      
```

Comment:
`NewRetrieved` seems to be only used for tests, so unclear why it is in this diagram, which I assume specifically is about not tests.

Reply:
This becomes a mute point with some other refactors on the repo: the Retrieved type here was just a helper for common functionality. Now the same is expressed as interfaces, `NewRetrieved` will disappear.